hax/treemakers/posrec.py

```diff
@@ -0,0 +1,200 @@
+from pax.plugins.interaction_processing.S1AreaFractionTopProbability import binom_test as binom_test_pax
+from keras.models import model_from_json
+import tensorflow as tf
+
+class PositionReconstruction(TreeMaker):
+    """Stores position-reconstruction-related variables.
+
+    Provides:
+       - s1_pattern_fit_hax: S1 pattern likelihood computed with corrected
+                             position and areas
+       - s1_pattern_fit_hits_hax: S1 pattern likelihood computed with corrected
+                                  position and hits
+       - s1_pattern_fit_bottom_hax: S1 pattern likelihood computed with corrected
+                                    position and bottom array area
+       - s1_pattern_fit_bottom_hits_hax: S1 pattern likelihood computed with corrected
+                                         position and bottom array hits
+
+       - s1_area_fraction_top_probability_hax: S1 AFT computed with corrected position
+
+       - x_observed_nn_tf: TensorFlow NN reconstructed x position
+       - y_observed_nn_tf: TensorFlow NN reconstructed y position
+       
+       - s1_area_upper_injection_fraction: s1 area fraction near Rn220 injection points (near PMT 131)
+       - s1_area_lower_injection_fraction: s1 area fraction near Rn220 injection points (near PMT 243)
+    """
+    __version__ = '0.15'
+    extra_branches = ['peaks.area_per_channel[260]',
+                      'peaks.hits_per_channel[260]',
+                      'peaks.n_saturated_per_channel[260]',
+                      'interactions.x', 'interactions.y', 'interactions.z']
+
+    def __init__(self):
+
+        hax.minitrees.TreeMaker.__init__(self)
+        
+        # We need to pull some stuff from the pax config
+        self.pax_config = load_configuration("XENON1T")
+        self.tpc_channels = list(range(0, 247 + 1))
+        self.confused_s1_channels = []
+        self.statistic = self.pax_config['BuildInteractions.BasicInteractionProperties'] \
+                         ['s1_pattern_statistic']
+        qes = np.array(self.pax_config['DEFAULT']['quantum_efficiencies'])
+
+        self.pattern_fitter = PatternFitter(
+            filename=utils.data_file_name(
+                self.pax_config['WaveformSimulator']['s1_patterns_file']),
+            zoom_factor=self.pax_config['WaveformSimulator'].get(
+                's1_patterns_zoom_factor', 1),
+            adjust_to_qe=qes[self.tpc_channels],
+            default_errors=(self.pax_config['DEFAULT']['relative_qe_error'] +
+                            self.pax_config['DEFAULT']['relative_gain_error'])
+        )
+
+        # For s1_aft variable
+        aftmap_filename = utils.data_file_name(hax.config["s1_aft_filename"])
+        self.aft_map = InterpolatingMap(aftmap_filename)
+        self.low_pe_threshold = 10
+
+        # load trained NN models
+        nn_model_json = utils.data_file_name(hax.config["neural_network_model"])
+        json_file_nn = open(nn_model_json, 'r')
```

Comment:
This is the bad way to load a model.  Please do `model.save('blah.hdf5')`, which will save the model and weights together. There's no reason to keep these apart and it can just lead to problems.

Reply:
Ah, there's also a keras.utils load_model command to reload it.